[Kaaviya Selvam] iP

kavxya opened the pull request:
# HelperBot  
>“Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)  

DukePro frees your mind of having to remember things you need to do. 
It's,  
-text-based 
-easy to learn 
-~~_FAST_~~ SUPER FAST to use  

All you need to do is,  
1.download it
2.double-click it. 
3.add your tasks. 
4.let it manage your tasks for you 😉  

And it is **FREE**!  
Features:   
-[x]Managing tasks  
-[ ]Managing deadlines (coming soon)  
-[ ]Reminders (coming soon)

Commit:
Add Gradle support

kavxya committed:
level 1 : skeleton

kavxya committed:
level 2

kavxya committed:
level 3: Mark as done

kavxya committed:
Level 4. ToDos, Events, Deadlines

kavxya committed:
testing

kavxya committed:
error

kavxya committed:
delete

kavxya committed:
Output list to duke.txt file

kavxya committed:
Read dates and store them in specified formate

kavxya committed:
Merge branch 'master' into branch-Level-7

kavxya committed:
Split main chuck of code into individual parts to emulate OOP

kavxya committed:
Package the classes

kavxya committed:
JUnit testing

kavxya committed:
Create jar

kavxya committed:
Add JavaDoc text to the files

kavxya committed:
Update to coding standard

kavxya committed:
Revert "Update to coding standard"

This reverts commit b2a1a65ee7100f74774ec18a73e87f8d018e245a.

kavxya committed:
Find tasks within the list

kavxya committed:
Merge branch 'branch-Level-9'

sheimoria commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,95 @@
+    /**
+     * Prints the error message for invalid commands.
+     */
+    public void doNotUnderstand() {
+        System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+    }
```
Should this method name be a verb instead?
```suggestion
    public void displayDoNotUnderstandError() {
        System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
    }
```

sheimoria commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,95 @@
+    /**
+     * Prints the message for adding a task to the taskList.
+     */
+    public void addTask(TaskList tasks) {
+        System.out.println("Got it. I've added this task: ");
+        System.out.println(tasks.get(tasks.size() - 1));
+        System.out.println(String.format("Now you have %d tasks in the list.", tasks.size()));
+
+    }
```
Might the method name be misleading? My first impression was that this method adds the task to the task list, but instead, it prints the task added.
```suggestion
    public void displayTaskAdded(TaskList tasks) {
        System.out.println("Got it. I've added this task: ");
        System.out.println(tasks.get(tasks.size() - 1));
        System.out.println(String.format("Now you have %d tasks in the list.", tasks.size()));

    }
```

sheimoria commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,51 @@
+        this.isDone = false;
+    }
+
+    /**
+     * To display the task to the user and data log.
+     * @return prints the task in the specified format
+     */
```
I think there should be an empty line between the description and parameter section as per the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#comments)

sheimoria commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+     * Returns a formatted string representing the Deadline task.
+     * @return String
+     */
+    public String toString() {
+        return String.format("[D][%s] %s", getStatusIcon(), description);
+    }
```
Maybe could consider adding the `@Override` annotation above all the `toString` methods? May not be necessary (coding standard does not talk about it) but I would think that it is recommended.
```suggestion
    @Override
    public String toString() {
        return String.format("[D][%s] %s", getStatusIcon(), description);
    }
```

sheimoria reviewed (commented):
LGTM, almost! Just a few nits to fix 👍

aachia commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,95 @@
+    /**
+     * Prints the message for marking a task as done.
+     *  @param task is the task which is marked as undone.
+     */
```
```suggestion
    /**
     * Prints the message for unmarking a task.
     *  @param task is the task which is unmarked.
     */
```

I think you might have a careless mistake here as the header does not describe correctly what the method does. In this case it is unmarking the task instead of marking.

aachia commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Changes the value of isDone to be false.
+     */
+    public void unmarkAsDone() {
```
```suggestion
    public void unmarkTask() {
```
Could this method name be confusing? From the original name, I would think that it unmarks a task and shows it as done which is conflicting.

kavxya committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d' into branch-A-CodingStandard

kavxya committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

kavxya committed:
Add Checkstyle support

kavxya committed:
Merge branch 'branch-A-CodingStandard'

kavxya committed:
add profile photos to Duke and the User

kavxya committed:
update gradle to the latest config

kavxya committed:
add Ui interface to duke

kavxya committed:
The current program does not make use of the Assert features.

Assert function to be added to the program to test certain hypothesis in the event of failure.

The number of characters for a list command should be fixed. The assert feature will verify the same and inform the user of any errors.

Lets,
Add the assert feature to the the Duke bot to ensure we are able to run the program smoothly.

kavxya committed:
Add the function to read the file from the storage

kavxya committed:
Update file loading from storage

kavxya committed:
Duke class:  skeleton framework of the program

Methods are factored out into super classes.

The common behaviors across the program is to parse the input from the user.

Extracting the common behavior into a super class allows us to use
polymorphism when dealing with multiple inputs.
For example same method can deal with various inputs.

Let's use SLAP to identify the method to be factored out, which is Parser in this instance

kavxya committed:
Revert "Duke class:  skeleton framework of the program"

This reverts commit f8caa813d7e5dc44f026ec76fb676a4b41250698.

kavxya committed:
Merge pull request #3 from kavxya/branch-CodeQuality

Changes to code quality

kavxya committed:
Merge pull request #2 from kavxya/branch-Assertions

Add assertions

kavxya committed:
Parser added to reduce chunks of code.

kavxya committed:
Merge branch 'branch-Assertions'

kavxya committed:
Merge commit '948c97f65414adff857222b62f80c9e179d67dc7'

kavxya committed:
Populate Ui with input and output commands

kavxya committed:
C-TagsAdd tags to the tasks

kavxya committed:
Set theme jekyll-theme-tactile

kavxya committed:
Ui Image

kavxya committed:
Merge branch 'master' of https://github.com/kavxya/ip

kavxya committed:
updates to userguide

kavxya committed:
Proper formatting of the list

kavxya committed:
Add specific clasifiers to the output the String output of the subclasses of Task.

kavxya committed:
Add interactive feedback for empty list.

Updates to the README.md file for formatting

kavxya committed:
Complete loading data from storage

kavxya committed:
updates to gradle

kavxya committed:
javafx

kavxya committed:
set source, build dependency to java 11

kavxya committed:
Refactoring.

kavxya committed:
README.md update

kavxya committed:
Refactoring

kavxya committed:
Fix data list bug.

kavxya committed:
JUnit Tests.
